insights/client/apps/compliance/__init__.py

```diff
@@ -27,63 +27,82 @@ def __init__(self, config):
 
     def oscap_scan(self):
         self._assert_oscap_rpms_exist()
-        policies = self.get_policies()
-        if not policies:
+        initial_profiles = self.get_initial_profiles()
+        matching_os_profiles = self.get_profiles_matching_os()
+        profiles = self.profile_union_by_ref_id(matching_os_profiles, initial_profiles)
+        if not profiles:
             logger.error("System is not associated with any profiles. Assign profiles using the Compliance web UI.\n")
             exit(constants.sig_kill_bad)
-        for policy in policies:
+        for profile in profiles:
             self.run_scan(
-                policy['attributes']['ref_id'],
-                self.find_scap_policy(policy['attributes']['ref_id']),
-                '/var/tmp/oscap_results-{0}.xml'.format(policy['attributes']['ref_id']),
-                tailoring_file_path=self.download_tailoring_file(policy)
+                profile['attributes']['ref_id'],
+                self.find_scap_policy(profile['attributes']['ref_id']),
+                '/var/tmp/oscap_results-{0}.xml'.format(profile['attributes']['ref_id']),
+                tailoring_file_path=self.download_tailoring_file(profile)
             )
 
         return self.archive.create_tar_file(), COMPLIANCE_CONTENT_TYPE
 
-    def download_tailoring_file(self, policy):
-        if 'tailored' not in policy['attributes'] or policy['attributes']['tailored'] is False:
+    def download_tailoring_file(self, profile):
+        if ('tailored' not in profile['attributes'] or profile['attributes']['tailored'] is False):
+                # or 'os_minor_version' not in profile['attributes'] or profile['attributes']['os_minor_version'] != self.os_minor_version()): # TODO after completion of COMP-E-133
             return None
 
         # Download tailoring file to pass as argument to run_scan
         logger.debug(
-            "Policy {0} is a tailored policy. Starting tailoring file download...".format(policy['attributes']['ref_id'])
+            "Policy {0} is a tailored policy. Starting tailoring file download...".format(profile['attributes']['ref_id'])
         )
-        tailoring_file_path = "/var/tmp/oscap_tailoring_file-{0}.xml".format(policy['attributes']['ref_id'])
+        tailoring_file_path = "/var/tmp/oscap_tailoring_file-{0}.xml".format(profile['attributes']['ref_id'])
```

Comment:
Unrelated to this idea, but this feels unsafe to construct path out of what comes from the API without any input validation. Mind to call basename on the formatted filename?

Reply:
Maybe we can revisit this in a future PR. I'm not sure how this could really be unsafe, since the file path is used to temporarily store the tailoring file so it can be passed to `oscap`. It doesn't exist before the client run, and trying to inject some bogus path here would just cause `oscap` to fail. Also, ref_id is immutable in our API, so it'd be impressive if a user could somehow hijack that value.

We should revisit the tmp dir that we use, though. There is `self.archive.tmp_dir`, which gets automatically cleaned up after the client run. Also, I think most of these paths should probably use `os.path.join` instead of just a string path. I'm going to hold off on these changes in this PR, since they're not related to the new functionality. If you find a genuine security threat, we should definitely report a jira and prioritize fixing it.

Reply:
Ack. It should be okay for the time being.

Here is what I had in mind, but it won't go through
```
$ touch /var/tmp/oscap_tailoring_file-/../test.xml
touch: cannot touch '/var/tmp/oscap_tailoring_file-/../test.xml': No such file or directory
```

We can further improve it as part of RHICOMPL-1196.